tests/files/acl/test-pre-wp-utils.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Automattic\VIP\Files\Acl\Pre_WP_Utils;
+
+use WP_Error;
+
+class VIP_Files_Acl_Pre_Wp_Utils_Test extends \WP_UnitTestCase {
+	public static function setUpBeforeClass() {
+		parent::setUpBeforeClass();
+
+		require_once( __DIR__ . '/../../../files/acl/pre-wp-utils.php' );
+	}
+
+	public function get_data__validate_path__invalid() {
+		return [
+			'null-uri' => [
+				null,
+				'VIP Files ACL failed due to empty path',
+			],
+	
+			'empty-uri' => [
+				'',
+				'VIP Files ACL failed due to empty path',
+			],
+	
+			'path-no-wp-content' => [
+				'/a/path/to/a/file.jpg',
+				'VIP Files ACL failed due to invalid path (for /a/path/to/a/file.jpg)',
+			],
+	
+			'relative-url-with-wp-content' => [
+				'en/wp-content/uploads/file.png',
+				'VIP Files ACL failed due to relative path (for en/wp-content/uploads/file.png)',
+			],
+		];
+	}
+	
+	public function get_data__validate_path__valid() {
+		return [
+			'valid-path' => [
+				'/wp-content/uploads/kittens.gif',
+			],
+	
+			'valid-path-nested' => [
+				'/wp-content/uploads/subfolder/2099/12/cats.jpg',
+			],
+	
+			'valid-path-subsite' => [
+				'/subsite/wp-content/uploads/puppies.png',
+			],
```

Comment:
Since this implementation supports it, can we add a test for a subsite that is several directory levels deep? `/foo/bar/baz/subsite/wp-content/uploads/puppies.png`

Reply:
Another test I just thought of - a URI that includes `wp-content/uploads` twice - `/wp-content/uploads/path/to/wp-content/uploads/otters.png`

Reply:
Added in 0590ad5e4